hal/src/stm32f2xx/usart_hal.c

```diff
@@ -547,8 +528,39 @@ void hal_usart_begin_config(hal_usart_interface_t serial, uint32_t baud, uint32_
 }
 
 void hal_usart_end(hal_usart_interface_t serial) {
-    usartMap[serial]->suspended = false;
     usartEndImpl(serial);
+
+    // Switch pins to INPUT
+    HAL_Pin_Mode(usartMap[serial]->rx_pin, INPUT);
```

Comment:
Nit: this seems almost identical to the code block in `hal_usart_sleep()`. Consider refactoring.

Reply:
Is there any defect configuring these pins to `INPUT` if UART is going to be ended? Or is it just violate the existing behavior?

Reply:
I'm just saying that you have code duplication :) We can leave things as-is or to make it a bit more concise to refactor into a common function with argument.